### PR TITLE
deps: Update yaml_cpp deps to fix CRNL and emit issue.

### PR DESCRIPTION
**Pull request by vehre-x41**
Commit Message: deps: Update yaml_cpp deps to fix CRNL and emit issues.
Additional Description: Update reference to yaml_cpp to make use of latest fixes:
- fix where on Windows a CRLF is not detected correctly
- emit of keys having special chars, like &, lead to illegal yaml.
Signed-off-by: Andre Vehreschild <vehre@x41-dsec.de>
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: Force yaml_cpp to link statically.

### CODE REVIEW

File: bazel/foreign_cc/yaml_cpp.patch
Comment (keith) [inline bazel/foreign_cc/yaml_cpp.patch:19]: Could this be in copts instead so these don't propagate up?

**Review by keith — COMMENTED** [review]
Are there upstream issues / PRs tracking these?

File: bazel/foreign_cc/yaml_cpp.patch
Comment (vehre-x41) [reply bazel/foreign_cc/yaml_cpp.patch:19]: They unfortunately need to propagate up or build on windows fails due to the API of yaml_cpp. I tried the copts in the beginning, but never to succeeded.

**Comment by vehre-x41** [comment]
> Are there upstream issues / PRs tracking these?

Er, what do you mean? This work is motivated by [PR20544](https://github.com/envoyproxy/envoy/pull/20544) not being the 'right' solution. The issue is in the yaml_cpp library and at least partially fixed there. Unfortunately way my fix not perfect, so have to iterate another patch to the yaml_cpp lib. This PR here is just for testing, whether the patch of yaml_cpp now really fixes all issues and builds on all platforms.

**Comment by keith** [comment]
Sorry I just meant for the newline one, so https://github.com/jbeder/yaml-cpp/pull/1099, can you add that as a comment in the patch so we know we can remove that once it merges?

**Comment by vehre-x41** [comment]
/donotmerge

This is a temporary state for testing purpose only. DO NOT MERGE!

File: bazel/repository_locations.bzl
Comment (vehre-x41) [reply bazel/repository_locations.bzl]: yaml-cpp has merged #1100. Updated refs and now just waiting for CI to pass.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
